packages/block-editor/src/components/block-variation-transforms/index.js

```diff
@@ -84,4 +94,82 @@ function __experimentalBlockVariationTransforms( { blockClientId } ) {
 	);
 }
 
+function __experimentalBlockVariationTransforms( { blockClientId } ) {
+	const { updateBlockAttributes } = useDispatch( blockEditorStore );
+	const { variations, blockAttributes, blockName } = useSelect(
+		( select ) => {
+			const { getBlockVariations } = select( blocksStore );
+			const { getBlockName, getBlockAttributes } = select(
+				blockEditorStore
+			);
+			const name = blockClientId && getBlockName( blockClientId );
+			return {
+				blockAttributes: getBlockAttributes( blockClientId ),
+				blockName: name,
+				variations: name && getBlockVariations( name, 'transform' ),
+			};
+		},
+		[ blockClientId ]
+	);
+	const { activeBlockVariation } = useSelect(
+		( select ) => {
+			const { getActiveBlockVariation } = select( blocksStore );
+			return {
+				activeBlockVariation: getActiveBlockVariation(
+					blockName,
+					blockAttributes
+				),
+			};
+		},
+		[ blockAttributes, blockName ]
+	);
```

Comment:
Could these two `useSelect` functions be merged? It seems like the second depends on the return value of the first anyway 🤷

Reply:
Good call! I tried a few different approaches so when I settled on this one, I missed that I could combine the two `useSelects`, thanks! I've updated in https://github.com/WordPress/gutenberg/pull/40036/commits/eebbba78b7d94c1ae2b38f12d75bfd8500d867d8